src/JSInterop/Microsoft.JSInterop/src/Implementation/JSObjectReferenceJsonWorker.cs

```diff
@@ -60,8 +68,24 @@ public static long ReadJSObjectReferenceIdentifier(ref Utf8JsonReader reader)
         public static void WriteJSObjectReference(Utf8JsonWriter writer, JSObjectReference objectReference)
         {
             writer.WriteStartObject();
-            writer.WriteNumber(_idKey, objectReference.Id);
+            writer.WriteNumber(_jsObjectIdKey, objectReference.Id);
             writer.WriteEndObject();
         }
     }
+
+    /// <summary>
+    /// Contains the JSObjectReference.Id and Length that was deserialized
+    /// </summary>
+    public struct DeserializedJSObjectReferenceValues
```

Comment:
Can this be made into a `readonly struct`?

TBH I find that collapsing the cases for "JS object reference" and "JS data reference" into a single type to be a bit artificial, given that there's no `Length` in the "object reference" case. Do you think it would work to handle them independently?

Reply:
BTW I know that would mean you have to either pass some "type" discriminator or use different ID keys to distinguish them.

Reply:
I believe b266fbafa8 should resolve this concern, please let me know if I may've misunderstood.